src/first.mjs

```diff
@@ -0,0 +1,7 @@
+import ensureIterable from './internal/ensure-iterable'
+
+export default function first (iterable) {
+  const iter = ensureIterable(iterable)[Symbol.iterator]()
+  const firstItem = iter.next()
```

Comment:
This gets a bit interesting. If the only usage of an iterable is taking the first item the underlying iterator needs to be notified, otherwise a leak occurs. Generators work this way -- they stay in a suspended execution state and AFAIK can't be garbage collected until they are finished yielding all their values. See: http://exploringjs.com/es6/ch_iteration.html#_closing-iterators-via-return

Come to think of it I think `zip` has this problem too, although not `zipLongest`.

You can easily do something like `if(typeof iter.return === 'function') { iter.return(); }`, but this would certainly be a break with `car`/`cdr` since using `first` would prevent you from then using `rest`.

Reply:
Completely unaware of this.
It probably impact many functions, I'll have a look into it

Reply:
I'm pretty certain it doesn't. I've audited the codebase for it before. I missed zipLongest when I did, but otherwise it's good. The case only affects manual iteration (calling `next()`) when the source iterable is not exhausted.